src/QsCompiler/Transformations/ContentLifting.cs

```diff
@@ -64,17 +63,22 @@ public CallableDetails(QsCallable callable)
 
         public class TransformationState
         {
-            public bool IsValidScope = true;
-            public List<QsCallable> GeneratedOperations = null;
-            public ImmutableArray<LocalVariableDeclaration<NonNullable<string>>> GeneratedOpParams =
+            // ToDo: It should be possible to make these three properties private, 
+            // if we absorb the corresponding logic into LiftBody. 
+            internal bool IsValidScope = true;
```

Comment:
Currently, this field needs to have the same access set as the whole LiftContent transformation, as the transformation could be inherited outside the assembly. In that case, that class would need access to this field to validate before calling LiftBody.

Reply:
Right. This is the one piece where I would suggest to look into whether there is a way to not have to expose that handle. Reading your replies on your PR possibly clarifies it a bit whether that makes sense. For this PR I'll make it public again, but it would be good if we could find a way around that.